src/Week12/Risk.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Week12.Risk where
+
+import Control.Monad.Random
+import Data.List
+
+------------------------------------------------------------
+-- Die values
+
+newtype DieValue = DV { unDV :: Int } 
+  deriving (Eq, Ord, Show, Num)
+
+first :: (a -> b) -> (a, c) -> (b, c)
+first f (a, c) = (f a, c)
+
+instance Random DieValue where
+  random           = first DV . randomR (1,6)
+  randomR (low,hi) = first DV . randomR (max 1 (unDV low), min 6 (unDV hi))
+
+die :: Rand StdGen DieValue
```

Comment:
In a real project you'd want to keep the type of this more polymorphic, so that it's more reusable
Something like `die :: MonadRandom m => m DieValue`

This comment applies to most other usages of `Rand StdGen` in this module too